var/spack/repos/builtin/packages/qt-creator/package.py

```diff
@@ -24,7 +25,8 @@ class QtCreator(QMakePackage):
     # built with a different version of sqlite than the bundled copy, it will
     # cause symbol conflict. Force Spack to build with the same version of
     # sqlite as the bundled copy.
-    depends_on('sqlite@3.8.10.2')
+    depends_on('sqlite@3.8.10.2', when='@:4.4.0')
+    depends_on('sqlite@3.8.10.3:')
```

Comment:
This needs a when clause, otherwise the versions will conflict. `depends_on('sqlite@3.8.10.3:', when='@4.8.0:')`

Reply:
fixed, thanks.